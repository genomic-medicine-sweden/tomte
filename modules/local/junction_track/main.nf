process JUNCTION_TRACK {
    tag 'junction_track'
    label 'process_single'

    conda "conda-forge::python=3.8.3"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/python:3.8.3' :
        'biocontainers/python:3.8.3' }"

    input:
    tuple val(meta), path(sj)

    output:
    tuple val(meta), path('*.bed'), emit: bed
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    $baseDir/bin/sj2bed.py --sj ${sj} --output ${prefix}_junction.bed

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        sj2bed: v1.0
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix    ?: "${meta.id}"
    """
    touch ${prefix}_junction.bed

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        sj2bed: v1.0
    END_VERSIONS
    """
}
