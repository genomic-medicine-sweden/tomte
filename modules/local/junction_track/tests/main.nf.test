nextflow_process {

    name "Test Process JUNCTION_TRACK"
    script "../main.nf"
    process "JUNCTION_TRACK"
    config "./nextflow.config"
    tag "modules_local"
    tag "modules"
    tag "junction_track"

    setup {
        run("STAR_GENOMEGENERATE") {
            script "../../../nf-core/star/genomegenerate/main.nf"
            process {
                """
                input[0] = Channel.of([
                        [ id:'test_fasta' ],
                        [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.fasta', checkIfExists: true) ]
                    ])
                    input[1] = Channel.of([
                        [ id:'test_gtf' ],
                        [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                    ])
                """
            }
        }

        run("STAR_ALIGN") {
            script "../../../nf-core/star/align/main.nf"
            process {
                """
                input[0] = Channel.of([
                    [ id:'test', single_end:true ], // meta map
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/illumina/fastq/test_rnaseq_1.fastq.gz', checkIfExists: true) ]
                ])
                input[1] = STAR_GENOMEGENERATE.out.index
                input[2] = Channel.of([
                    [ id:'test_gtf' ],
                    [ file(params.modules_testdata_base_path + 'genomics/homo_sapiens/genome/genome.gtf', checkIfExists: true) ]
                ])
                input[3] = false
                input[4] = 'illumina'
                input[5] = false
                """
            }
        }
    }


    test("Should run without failures") {

        when {
            params {
                modules_testdata_base_path = 'https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/'
            }
            process {
                """
                input[0] = STAR_ALIGN.out.spl_junc_tab
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
