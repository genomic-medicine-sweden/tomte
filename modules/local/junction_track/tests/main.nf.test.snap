{
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_junction.bed:md5,81f54c1735790bdbf89b55d76b2818e8"
                    ]
                ],
                "1": [
                    "versions.yml:md5,1188acb4f364f95c513ffeb026b17be0"
                ],
                "bed": [
                    [
                        {
                            "id": "test",
                            "single_end": true
                        },
                        "test_junction.bed:md5,81f54c1735790bdbf89b55d76b2818e8"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,1188acb4f364f95c513ffeb026b17be0"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-04-16T10:31:57.222315"
    }
}